src/sv_port.rs

```diff
@@ -204,11 +208,82 @@ fn port_signedness_ansi(m: &sv_parser::AnsiPortDeclaration) -> SvSignedness {
     }
 }
 
+fn port_packeddim_ansi(
+    m: &sv_parser::AnsiPortDeclaration,
+    syntax_tree: &SyntaxTree,
+) -> Vec<SvPackedDimension> {
+    let mut ret: Vec<SvPackedDimension> = Vec::new();
+
+    let mut upper = String::new();
+    let mut lower = String::new();
+
+    for node in m {
+        match node {
+            RefNode::PackedDimension(x) => {
+                let mut div_found: bool = false;
+                for sub_node in x {
+                    match sub_node {
+                        RefNode::Symbol(x) => {
+                            if syntax_tree.get_str(&x.nodes.0).unwrap().to_string() == ":" {
```

Comment:
Can you use ConstantRange instead of manually looking for the colon?
https://docs.rs/sv-parser/latest/sv_parser/struct.ConstantRange.html
I was expecting something like just converting ConstantExpression to a string.

Reply:
I just updated it to be based on ConstantRange.